include/oneapi/dpl/internal/binary_search_impl.h

```diff
@@ -83,43 +83,49 @@ struct custom_brick
 template <class _Tag, typename Policy, typename InputIterator1, typename InputIterator2, typename OutputIterator,
           typename StrictWeakOrdering>
 OutputIterator
-lower_bound_impl(_Tag, Policy&& policy, InputIterator1 start, InputIterator1 end, InputIterator2 value_start,
+lower_bound_impl(_Tag tag, Policy&& policy, InputIterator1 start, InputIterator1 end, InputIterator2 value_start,
                  InputIterator2 value_end, OutputIterator result, StrictWeakOrdering comp)
 {
     static_assert(__internal::__is_host_dispatch_tag_v<_Tag>);
 
-    return oneapi::dpl::transform(policy, value_start, value_end, result,
-                                  [=](typename ::std::iterator_traits<InputIterator2>::reference val) {
-                                      return ::std::lower_bound(start, end, val, comp) - start;
-                                  });
+    using _ValueType = typename std::iterator_traits<InputIterator2>::value_type;
```

Comment:
What is this for? Why doesn't `traits::reference` work in lambda parameter?

Reply:
Sure, It's works too.
It was suggestion from @MikeDvorskiy : https://github.com/oneapi-src/oneDPL/pull/1846#pullrequestreview-2306891807

"
As far as we are changing this place... Lets changes 'val' type as well, according to 
https://en.cppreference.com/w/cpp/algorithm/lower_bound, to `const T& value`
Also it is relevant to `upper_bound_impl` and `binary_search_impl`.
"

Reply:
I remember now. Thanks Mikhail.